Type error after updatin to Angular 15

I'm getting following errors after updating to Angular 15:
```
Error: node_modules/@ngx-resource/core/lib/Declarations.d.ts:110:125 - error TS2344: Type 'O' does not satisfy the constraint '{}'.

110 export interface IResourceMethodResultStrict<IB, IQ, IP, O> extends IResourceMethodStrictBase<IB, IQ, IP, O, ResourceResult<O>> {

  node_modules/@ngx-resource/core/lib/Declarations.d.ts:110:58
    110 export interface IResourceMethodResultStrict<IB, IQ, IP, O> extends IResourceMethodStrictBase<IB, IQ, IP, O, ResourceResult<O>> {
    This type parameter might need an `extends {}` constraint.

Error: node_modules/@ngx-resource/core/lib/Declarations.d.ts:112:97 - error TS2344: Type 'O' does not satisfy the constraint '{}'.

112 export interface IResourceMethodResult<IB, O> extends IResourceMethodBase<IB, O, ResourceResult<O>> {

  node_modules/@ngx-resource/core/lib/Declarations.d.ts:112:44
    112 export interface IResourceMethodResult<IB, O> extends IResourceMethodBase<IB, O, ResourceResult<O>> {
    This type parameter might need an `extends {}` constraint.
```

Most likely this is because of typescript version changed on that Angular release. Any idea how to solve this issue?

--- a/projects/ngx-resource/core/src/lib/Declarations.ts
+++ b/projects/ngx-resource/core/src/lib/Declarations.ts
@@ -154,11 +154,11 @@ export interface IResourceMethodObservable<IB, O>
 }
 
 
-export interface IResourceMethodResultStrict<IB, IQ, IP, O>
+export interface IResourceMethodResultStrict<IB, IQ, IP, O extends {}>
   extends IResourceMethodStrictBase<IB, IQ, IP, O, ResourceResult<O>> {
 }
 
-export interface IResourceMethodResult<IB, O>
+export interface IResourceMethodResult<IB, O extends {}>
   extends IResourceMethodBase<IB, O, ResourceResult<O>> {
 }
 

